hack/lib/log/output.sh

```diff
@@ -1,27 +1,52 @@
 #!/bin/bash
 
 # This file contains functions used for writing log messages
-# to stdout and stderr from scripts while they run.
-
-function os::log::with-severity() {
-	local msg=$1
-	local severity=$2
-
-	echo "[$2] ${1}"
-}
-readonly -f os::log::with-severity
+# to stdout and stderr from scripts while they run. `printf`
+# is used in order to support newlines in the output while
+# correctly showing literal escape characters that exist in
+# the mesage being logged. `echo -e` would instead interpret
+# literal escapes in the logged message.
 
+# os::log::info writes the message to stdout
+#
+# Arguments:
+#  - all: message to write
 function os::log::info() {
-	os::log::with-severity "${1}" "INFO"
+	printf '[INFO] %s\n' "$*"
 }
 readonly -f os::log::info
 
+# os::log::warn writes the message to stderr.
+# A warning indicates something went wrong but
+# not so wrong that we cannot recover.
+#
+# Arguments:
+#  - all: message to write
 function os::log::warn() {
-	os::log::with-severity "${1}" "WARNING" 1>&2
+	os::text::print_yellow "$( printf '[WARNING] %s\n' "$*" )" 1>&2
```

Comment:
Hmm... the intent was to try to keep the output of `$*` as true as possible to what was passed in to the function, while also allowing for a trailing newline to be added and literal escapes in the input message to be left un-parsed. I figured `printf` would leave the input message as it were while `echo` may interpret escapes. Is this unfounded? I do like `os::text::print_yellow "[WARNING] $*" 1>&2;`